src/mongo_client.ts

```diff
@@ -364,6 +364,47 @@ export class MongoClient extends TypedEventEmitter<MongoClientEvents> {
         return true;
       }
     };
+    this.checkForNonGenuineHosts();
+  }
+
+  /** @internal */
+  private checkForNonGenuineHosts() {
+    if (this.mongoLogger && this.mongoLogger.logDestination) {
+      const documentDBHostnames = this[kOptions].hosts.filter(
+        (hostAddress: HostAddress) =>
+          hostAddress.host &&
```

Comment:
I'd like to propose a few things here.

First, I think we should extract all the log messages to constants so we're not instantiating the same strings over and over again with each instantiation of a `MongoClient`, and since they are not dynamic.

Then we've got a pattern here of how we check if a host matches a specific domain, so we can extract a method to handle that common behaviour. And we can simplify that behaviour by using a Regex instead of potential multiple `endsWith` calls.

Example:
```typescript
const DOCUMENT_DB_CHECK = /(\.docdb\.amazonaws\.com$)|(\.docdb-elastic\.amazonaws\.com$)/
const COSMOS_DB_CHECK = /\.cosmos\.azure\.com$/

function isHostMatch(match: RegExp, host?: string): boolean {
  return host && match.test(host);
}
```

Then in the code we could refactor to do something like:

```typescript
const documentDbHostnames = this[kOptions].hosts.filter(
  (hostAddress: HostAddress) => isMostMatch(DOCUMENT_DB_CHECK, hostAddress)
);
const srvHostIsDocumentDB = isHostMatch(DOCUMENT_DB_CHECK, this[kOptions].srvHost);
```

Thoughts?

Reply:
I agree with both suggestions, I'll implement these changes.

Reply:
What are your thoughts on having Document_DB_Check and Cosmos_DB_Check as static class variables of MongoClient? Since they won't be used outside of the class, and classes cannot have const variables in JS.

Reply:
See other comment, I think they should actually be moved.